CMakeLists.txt

```diff
@@ -207,6 +207,19 @@ cmake_dependent_option(
 option(USE_TBB "Use TBB" OFF)
 option(ONNX_ML "Enable traditional ONNX ML API." ON)
 
+# Linux distributions do not want too many embedded sources
+option(USE_SYSTEM_LIBS "Use system-provided libraries." OFF)
+option(USE_SYSTEM_CPUINFO "Use system-provided cpuinfo." OFF)
+option(USE_SYSTEM_SLEEF "Use system-provided sleef." OFF)
+if(DEFINED ENV{USE_SYSTEM_LIBS})
+  set(USE_SYSTEM_LIBS "$ENV{USE_SYSTEM_LIBS}")
+endif()
+if(USE_SYSTEM_LIBS)
+  set(USE_SYSTEM_CPUINFO ON)
+  set(USE_SYSTEM_SLEEF ON)
+  set(BUILD_CUSTOM_PROTOBUF OFF)
+endif()
```

Comment:
@xuhdev would you mind checking this and saying if `cmake_dependent_option` would be appropriate here?

Reply:
If `USE_SYSTEM_LIBS` is `OFF`, does it mean the three options won't be available? If so, I think `cmake_dependent_option` is appropriate.

Reply:
USE_SYSTEM_LIBS is a shortcut variable to toggle all the USE_SYSTEM_* variables on. Users can still toggle e.g. USE_SYSTEM_SLEEF without turning USE_SYSTEM_LIBS on. I'll write it to the comments in the next update.